src/viam/examples/modules/mlmodelservice_tflite.cpp

```diff
@@ -0,0 +1,809 @@
+        // `TfLiteModelCreate`. That still requires that the buffer be
+        // kept valid, but that's more easily done.
+        std::ifstream in(*model_path_string, std::ios::in | std::ios::binary);
+        if (!in) {
+            std::ostringstream buffer;
+            buffer << service_name << ": Failed to open file for `model_path` "
+                   << model_path_string;
+            throw std::invalid_argument(buffer.str());
+        }
+        std::ostringstream model_path_contents_stream;
+        model_path_contents_stream << in.rdbuf();
+        state->model_data = std::move(model_path_contents_stream.str());
+
+        // Create an error reporter so that we can extract detailed
+        // error information from TFLite when things go wrong. The
+        // error state is protected by the interpreter lock.
+        state->model.reset(TfLiteModelCreateWithErrorReporter(
+            state->model_data.data(),
+            state->model_data.size(),
+            [](void* ud, const char* fmt, va_list args) {
+                char buffer[4096];
+                static_cast<void>(vsnprintf(buffer, sizeof(buffer), fmt, args));
+                *reinterpret_cast<std::string*>(ud) = buffer;
+            },
+            &state->interpreter_error_data));
+
+        // If we failed to create the model, return an error and
+        // include the error data that tflite wrote to the error
+        // reporter state.
+        if (!state->model) {
+            std::ostringstream buffer;
+            buffer << service_name << ": Failed to load model from file `" << model_path_string
+                   << "`: " << state->interpreter_error_data;
+            throw std::invalid_argument(buffer.str());
+        }
+
+        // If present, extract and validate the number of threads to
+        // use in the interpreter and create an interpreter options
+        // object to carry that information.
+        auto num_threads = attributes->find("num_threads");
+        if (num_threads != attributes->end()) {
+            const auto* num_threads_double = num_threads->second->get<double>();
+            if (!num_threads_double || !std::isnormal(*num_threads_double) ||
+                (*num_threads_double < 0) ||
+                (*num_threads_double >= std::numeric_limits<std::int32_t>::max()) ||
+                (std::trunc(*num_threads_double) != *num_threads_double)) {
+                std::ostringstream buffer;
+                buffer << service_name
+                       << ": Value for field `num_threads` is not a positive integer: "
+                       << *num_threads_double;
+                throw std::invalid_argument(buffer.str());
+            }
+
+            state->interpreter_options.reset(TfLiteInterpreterOptionsCreate());
+            TfLiteInterpreterOptionsSetNumThreads(state->interpreter_options.get(),
+                                                  static_cast<int32_t>(*num_threads_double));
```

Comment:
Should we set a limit to number of threads someone can request? In Go, we warn the user that we limited their threads to maxCPUs/2 if they requested a larger amount.

Reply:
Sounds good to me! I'm good with leaving things as is